docs/data_sharing_guide.md

```diff
@@ -0,0 +1,338 @@
+# Data Sharing Guide: CDM JupyterHub
+
+## Overview
+
+The BER DataLake JupyterHub platform provides data sharing capabilities that allow users to share their datasets with other users and groups. 
+
+All data governance functions are **✨ automatically imported** in every notebook via `startup.py` - no manual imports needed! This guide shows you how to use these pre-loaded functions for seamless data sharing and collaboration.
+
+## Key Concepts
+
+### Data Organization
+
+All user data in BER DataLake is organized under personal namespaces:
+
+- **General Data Storage**: `s3a://cdm-lake/users-general-warehouse/{username}/`
+- **SQL Warehouse**: `s3a://cdm-lake/users-sql-warehouse/{username}/`
+
+## Getting Started
+
+### Auto-Imported Functions
+
+All CDM JupyterHub notebooks automatically import these data governance functions at startup (via `startup.py`), so they're ready to use immediately without any imports:
+
+**Auto-Imported Utility Functions:**
+- `check_governance_health()` - Check service status
+- `get_minio_credentials()` - Get your MinIO credentials (sets environment variables)
+- `get_my_sql_warehouse()` - Get your SQL warehouse prefix
+- `get_my_policies()` - Get detailed policy information
+- `get_my_workspace()` - Get comprehensive workspace information
+- `get_path_access_info(path)` - Check who has access to a path
+- `make_table_private(namespace, table_name)` - Remove public access from SQL tables
+- `make_table_public(namespace, table_name)` - Make SQL tables publicly accessible
+- `share_table(namespace, table_name, with_users, with_groups)` - Share SQL tables
+- `unshare_table(namespace, table_name, from_users, from_groups)` - Unshare SQL tables
```

Comment:
I'm still pretty sure that although this will make the tables visible via Minio, a user won't be able to interact with another user's tables since they're in a different SQL warehouse. They'll be able to see them in Hive but IIUC any reads or writes will go to their own warehouse. Happy to be proven wrong, that'd be great.

Reply:
created ticket - https://kbase-jira.atlassian.net/jira/software/projects/CDM/boards/88/backlog?selectedIssue=CDM-315

Reply:
https://kbase-jira.atlassian.net/browse/CDM-315 is the direct link fwiw

:+1: